packages/php/blueprint/src/Importers/ImportInstallTheme.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Automattic\WooCommerce\Blueprint\Importers;
+
+use Automattic\WooCommerce\Blueprint\ResourceStorages;
+use Automattic\WooCommerce\Blueprint\StepProcessor;
+use Automattic\WooCommerce\Blueprint\StepProcessorResult;
+use Automattic\WooCommerce\Blueprint\Steps\InstallTheme;
+use Automattic\WooCommerce\Blueprint\UseWPFunctions;
+use Plugin_Upgrader;
+
+class ImportInstallTheme implements StepProcessor {
+	use UseWPFunctions;
+	private ResourceStorages $storage;
+	private StepProcessorResult $result;
+
+	public function __construct( ResourceStorages $storage ) {
+		$this->result  = StepProcessorResult::success( 'InstallThemes' );
+		$this->storage = $storage;
+	}
+	public function process( $schema ): StepProcessorResult {
+		$installed_themes = wp_get_themes();
+		$theme            = $schema->themeZipFile;
+
+		if ( isset( $installed_themes[ $theme->slug ] ) ) {
+			$this->result->add_info( "Skipped installing {$theme->slug}. It is already installed." );
+			return $this->result;
+		}
+		if ( $this->storage->is_supported_resource( $theme->resource ) === false ) {
+			$this->result->add_error( "Invalid resource type for {$theme->slug}" );
+			return $this->result;
+		}
+
+		$downloaded_path = $this->storage->download( $theme->slug, $theme->resource );
+
+		if ( ! $downloaded_path ) {
+			$this->result->add_error( "Unable to download {$theme->slug} with {$theme->resource} resource type." );
+			return $this->result;
+		}
+
+		$this->result->add_debug( "'$theme->slug' has been downloaded in $downloaded_path" );
+
+		$install = $this->install( $downloaded_path );
+		$install && $this->result->add_debug( "Theme '$theme->slug' installed successfully." );
```

Comment:
Should we still produce debug message if install failed?

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/49763/commits/0704d539d00323b7a648d3dc4b33829b3ec2f767#diff-21cc57aba191d958e0c9ba83a2b371cc15e9b8437b2428135810bcf87705aaf7R48